File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        int width = 0;
        int height = 0;
        int total = brown + yellow;
        
        height = (brown+4)/4;
        width = (brown+4) / 2 - height;
        
        while(true) {
            if(total == width * height) {
                answer[0] = width;
                answer[1] = height;
                break;
            }
            width++;
            height--;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    
    let width = 0;
    let height = 0;
    let total = brown + yellow;
    
    height = Math.floor((brown+4)/4);
    width = (brown+4)/2 - height;

    while(true){
        
        if(total === width * height){
            answer[0] = width;
            answer[1] = height;
            break;
        }
        width++;
        height--;
    }
    
    return answer;
}